library/lib_navball.ks

```diff
@@ -11,44 +11,111 @@ function east_for {
 }
 
 function compass_for {
-  parameter ves.
+  parameter ves,thing is "default".
 
   local pointing is ves:facing:forevector.
+  if not thing:istype("string") {
+    set pointing to type_to_vector(ves,thing).
+  }
+
   local east is east_for(ves).
 
   local trig_x is vdot(ves:north:vector, pointing).
   local trig_y is vdot(east, pointing).
 
   local result is arctan2(trig_y, trig_x).
 
-  if result < 0 { 
+  if result < 0 {
     return 360 + result.
   } else {
     return result.
   }
 }
 
 function pitch_for {
-  parameter ves.
+  parameter ves,thing is "default".
+
+  local pointing is ves:facing:forevector.
+  if not thing:istype("string") {
+    set pointing to type_to_vector(ves,thing).
+  }
 
-  return 90 - vang(ves:up:vector, ves:facing:forevector).
+  return 90 - vang(ves:up:vector, pointing).
 }
 
 function roll_for {
-  parameter ves.
-  
-  if vang(ves:facing:vector,ves:up:vector) < 0.2 { //this is the dead zone for roll when the vessel is vertical
+  parameter ves,thing is "default".
+
+  local pointing is ves:facing.
+  if not thing:istype("string") {
+    if thing:istype("vessel") or pointing:istype("part") {
+      set pointing to thing:facing.
+    } else if thing:istype("direction") {
+      set pointing to thing.
+    } else {
+      print "type: " + thing:typename + " is not reconized by lib_navball".
```

Comment:
```suggestion
      print "type: " + thing:typename + " is not reconized by roll_for".
```
This is specific for this function only.

Reply:
done